Source/WebCore/loader/FrameLoader.cpp

```diff
@@ -1273,6 +1273,15 @@ void FrameLoader::loadInSameDocument(URL url, RefPtr<SerializedScriptValue> stat
 
     checkedHistory()->updateForSameDocumentNavigation();
 
+    if (document->settings().navigationAPIEnabled() && !url.isAboutBlank()) {
+        if (RefPtr domWindow = document->domWindow()) {
+            if (RefPtr currentItem = checkedHistory()->currentItem()) {
+                auto item = currentItem.releaseNonNull();
```

Comment:
This probably doesn't need its own line.

Reply:
I don't think that is possible here since we need to hand over a Ref<> &.

Reply:
I think we should hand over a Ref&&